extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactivePubSubCommandsImpl.java

```diff
@@ -166,6 +195,24 @@ public Uni<String> subscribe() {
                         }
                     }
                 });
+                if (onEndHandler != null) {
+                    connection.endHandler(() -> {
+                        if (Vertx.currentContext() != null) {
+                            onEndHandler.run();
+                        } else {
+                            datasource.getVertx().runOnContext(onEndHandler);
```

Comment:
Might need to create a duplicated context here

Reply:
Something like this?

    connection.endHandler(() -> {
        Context ctxt = Vertx.currentContext();
        if (ctxt != null) {
            VertxContext.getOrCreateDuplicatedContext(ctxt)
                                 .runOnContext(ignored -> onEndHandler.run());
        } else {
            datasource.getVertx().runOnContext(() -> {
                VertxContext.getOrCreateDuplicatedContext(ctxt)
                                    .runOnContext(ignored -> onEndHandler.run());
            });
        }
    });

Reply:
The created duplicated context should also be marked as safe (or you won't be able to use hibernate reactive)

Reply:
How to mark it as safe?